R/proxy.R

```diff
@@ -180,6 +179,10 @@ proxy <- function(x, y = NULL, margin = 1,
         x <- as(as(x, "lMatrix"), "dMatrix")
         y <- as(as(y, "lMatrix"), "dMatrix")
     }
+    threads <- as.integer(getOption("proxyC.threads", -1))
+    if (length(threads) != 1 || is.na(threads)) {
+        stop("proxyC.threads must be an integer")
+    }
```

Comment:
I'd be inclined to add a helper function, something along the lines of

```r
getThreads <- function() {
  threads <- getOption("proxyC.threads", Sys.getenv("RCPP_PARALLEL_NUM_THREADS"))
  if (is.null(threads)) {
    threads <- -1L
  }
  threads <- as.integer(threads)
  if (length(threads != 1) || is.na(threads)) {
    stop("proxyC.threads must be an integer")
  }
  threads
}
```

I'd propose still supporting `RCPP_PARALLEL_NUM_THREADS` as an environment variable because otherwise this PR is a breaking change.

Reply:
That makes sense. I added `getThreads()` that also takes value from `OMP_THREAD_LIMIT` to stop using too many threads in CRAN tests.